### PR TITLE
Add recommendations placeholder page, component, fixtures, and delete

### LINKED ISSUE #14: Add Delete Column to RecommendationTable (if not already done)
- [x] Create a delete column with a Delete Button in each row
- [x] When clicked, deletes row from database and visually from the table
- [x] There is full test coverage (Jacoco) for RecommendationRequestsController
- [x] There is full mutation test coverage (Pitest) for new code

### PR DESCRIPTION
**Pull request by TokyoExpress**
Added a placeholder for the RecommendationsMenuItemIndex Page. Updated the routing in App.js to redirect to the placeholder page. Added tests.

Added components and fixtures for RecommendationTable. Added delete column at this step. Replaced index page. Added tests.

100% coverage with npm run coverage
Mutation score of 100 with npx stryker run

Issues https://github.com/ucsb-cs156-s22/team03-s22-5pm-3/issues/10 https://github.com/ucsb-cs156-s22/team03-s22-5pm-3/issues/12 https://github.com/ucsb-cs156-s22/team03-s22-5pm-3/issues/13 https://github.com/ucsb-cs156-s22/team03-s22-5pm-3/issues/14

### CODE REVIEW

**Review by ryang1337 — CHANGES_REQUESTED** [review]
Have you tested your code on localhost yet? It doesn't seem to load the table when you click on the link.

File: frontend/src/App.js [hunk 1]
```diff
@@ -71,6 +72,13 @@ function App() {
             </>
           )
         }
+        {
+          hasRole(currentUser, "ROLE_USER") && (
+            <>
+              <Route exact path="/recommendations/list" element={<RecommendationsIndexPage />} />
```
Comment (yuheho7749) [inline frontend/src/App.js]: /recommendationrequests/list

File: frontend/src/main/components/Recommendations/RecommendationsTable.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import OurTable, { ButtonColumn} from "main/components/OurTable";
+import { useBackendMutation } from "main/utils/useBackend";
+import {  onDeleteSuccess } from "main/utils/RecommendationsUtils"
+// import { useNavigate } from "react-router-dom";
+import { hasRole } from "main/utils/currentUser";
+
+
+export function cellToAxiosParamsDelete(cell) {
+    return {
+        url: "/api/recommendations",
```
Comment (yuheho7749) [inline frontend/src/main/components/Recommendations/RecommendationsTable.js]: /api/recommendationrequests

File: frontend/src/main/components/Recommendations/RecommendationsTable.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    // const editCallback = (cell) => {
+    //     navigate(`/ucsbdates/edit/${cell.row.values.id}`)
+    // }
+
+    // Stryker disable all : hard to test for query caching
+    const deleteMutation = useBackendMutation(
+        cellToAxiosParamsDelete,
+        { onSuccess: onDeleteSuccess },
+        ["/api/recommendations/all"]
```
Comment (yuheho7749) [inline frontend/src/main/components/Recommendations/RecommendationsTable.js]: /api/recommendationrequests

File: frontend/src/main/pages/Recommendations/RecommendationsIndexPage.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import { useCurrentUser } from 'main/utils/currentUser' // use prefix indicates a React Hook
+
+export default function RecommendationsIndexPage() {
+
+  const currentUser = useCurrentUser();
+
+  const { data: recommendations, error: _error, status: _status } =
+    useBackend(
+      // Stryker disable next-line all : don't test internal caching of React Query
+      ["/api/recommendations/all"],
```
Comment (yuheho7749) [inline frontend/src/main/pages/Recommendations/RecommendationsIndexPage.js]: /api/recommendationrequests

File: frontend/src/main/pages/Recommendations/RecommendationsIndexPage.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+            // Stryker disable next-line StringLiteral,ObjectLiteral : since "GET" is default, "" is an equivalent mutation
+            { method: "GET", url: "/api/recommendations/all" },
```
Comment (yuheho7749) [inline frontend/src/main/pages/Recommendations/RecommendationsIndexPage.js]: /api/recommendationrequests

File: frontend/src/main/utils/RecommendationsUtils.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { toast } from "react-toastify";
+
+export function onDeleteSuccess(message) {
+    console.log(message);
+    toast(message);
+}
+
+export function cellToAxiosParamsDelete(cell) {
+	return {
+        url: "/api/recommendations",
```
Comment (yuheho7749) [inline frontend/src/main/utils/RecommendationsUtils.js]: /api/recommendationrequests

File: frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+        axiosMock.reset();
+        axiosMock.resetHistory();
+        axiosMock.onGet("/api/currentUser").reply(200, apiCurrentUserFixtures.adminUser);
+        axiosMock.onGet("/api/systemInfo").reply(200, systemInfoFixtures.showingNeither);
+    };
+
+    test("renders without crashing for regular user", () => {
+        setupUserOnly();
+        const queryClient = new QueryClient();
+        axiosMock.onGet("/api/recommendations/all").reply(200, []);
```
Comment (yuheho7749) [inline frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js]: /api/recommendationrequests

File: frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+            </QueryClientProvider>
+        );
+
+
+    });
+
+    test("renders without crashing for admin user", () => {
+        setupAdminUser();
+        const queryClient = new QueryClient();
+        axiosMock.onGet("/api/recommendations/all").reply(200, []);
```
Comment (yuheho7749) [inline frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js]: /api/recommendationrequests

File: frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+            </QueryClientProvider>
+        );
+
+
+    });
+
+    test("renders three recommendations without crashing for regular user", async () => {
+        setupUserOnly();
+        const queryClient = new QueryClient();
+        axiosMock.onGet("/api/recommendations/all").reply(200, recommendationsFixtures.threeRecommendations);
```
Comment (yuheho7749) [inline frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js]: /api/recommendationrequests

File: frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+        await waitFor(  () => { expect(getByTestId(`${testId}-cell-row-0-col-requesterEmail`)).toHaveTextContent("supbub@gmail.com"); } );
+        expect(getByTestId(`${testId}-cell-row-1-col-requesterEmail`)).toHaveTextContent("timetravel@gmail.com");
+        expect(getByTestId(`${testId}-cell-row-2-col-requesterEmail`)).toHaveTextContent("loltown@gmail.com");
+
+    });
+
+    test("renders three recommendations without crashing for admin user", async () => {
+        setupAdminUser();
+        const queryClient = new QueryClient();
+        axiosMock.onGet("/api/recommendations/all").reply(200, recommendationsFixtures.threeRecommendations);
```
Comment (yuheho7749) [inline frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js]: /api/recommendationrequests

File: frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+        expect(getByTestId(`${testId}-cell-row-1-col-requesterEmail`)).toHaveTextContent("timetravel@gmail.com");
+        expect(getByTestId(`${testId}-cell-row-2-col-requesterEmail`)).toHaveTextContent("loltown@gmail.com");
+
+    });
+
+    test("renders empty table when backend unavailable, user only", async () => {
+        setupUserOnly();
+
+        const queryClient = new QueryClient();
+        axiosMock.onGet("/api/recommendations/all").timeout();
```
Comment (yuheho7749) [inline frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js]: /api/recommendationrequests

File: frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+        });
+
+        expect(queryByTestId(`${testId}-cell-row-0-col-requesterEmail`)).not.toBeInTheDocument();
+    });
+
+    test("test what happens when you click delete, admin", async () => {
+        setupAdminUser();
+
+        const queryClient = new QueryClient();
+        axiosMock.onGet("/api/recommendations/all").reply(200, recommendationsFixtures.threeRecommendations);
```
Comment (yuheho7749) [inline frontend/src/tests/pages/Recommendations/RecommendationsIndexPage.test.js]: /api/recommendationrequests

File: frontend/src/tests/utils/RecommendationsUtils.test.js [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+        test("It returns the correct params", () => {
+            // arrange
+            const cell = { row: { values: { requesterEmail: "supbub@gmail.com" } } };
+
+            // act
+            const result = cellToAxiosParamsDelete(cell);
+
+            // assert
+            expect(result).toEqual({
+                url: "/api/recommendations",
```
Comment (yuheho7749) [inline frontend/src/tests/utils/RecommendationsUtils.test.js]: /api/recommendationrequests

**Review by yuheho7749 — CHANGES_REQUESTED** [review]
just need to fix the routing urls

**Comment by TokyoExpress** [comment]
fixed routing errors

File: frontend/src/main/components/Recommendations/RecommendationsTable.js [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+            Header: 'Requester Email',
+            accessor: 'requesterEmail', 
+        },
+        {
+            Header: 'Professor Email',
+            accessor: 'professorEmail',
+        },
+        {
+            Header: 'Explanation',
+            id: 'explanation',
```
Comment (yuheho7749) [inline frontend/src/main/components/Recommendations/RecommendationsTable.js:48]: Hint: this line needs changes

**Review by yuheho7749 — CHANGES_REQUESTED** [review]
Please test on your local machine/host!!!!!!!!!!!!!!!!!!!!! @TokyoExpress

**Pull request closed without merging**